File: cyclone_word.py
def is_cyclone_word(word):
    # Convert the word to lowercase for case-insensitive comparison
    word = word.lower()
    
    # Check if the word has at least 3 characters
    if len(word) < 3:
        return False
    
    # Create two pointers for cyclic pattern comparison
    left = 0
    right = len(word) - 1
    
    while left < right: # left kieu gi cx nho hon right (tru truong hop tu co 2 chu)
        # Check if the characters are not in alphabetical order
        if word[left] > word[right]:
            return False
        
        # Move the pointers in a cyclic pattern
        left += 1
        right -= 1
        
        # Skip non-alphabetic characters
        while not word[left].isalpha():
            left += 1
        while not word[right].isalpha():
            right -= 1
    
    return True

def main():
    word = "settled"
    result = is_cyclone_word(word)
    print(f"{word} is a cyclone word: {result}")

if __name__ == "__main__":
    main()

File: cyclone_word.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int is_cyclone_word(char* word){
    int len = strlen(word);
    printf("%d\n", len);
    int left = 0;
    int right = len-1;

    while (left < right)
    {   
        if (word[left] > word[right]){
            return 0;
        }
        left += 1;
        right -=1;
    }

    left=1; right = len-1;
    while (left < right)
    {
        if (word[left] < word[right]){
            return 0;
        }
        left += 1;
        right -=1;
    }


    return 1;
}

// 0, n, 1, n-1, 2, n-2 
/* a=1, b = n (index)
left=w[a], right = w[b]
a=1, b=n
left<right (1,n) a+=1, b-=1
left<right (2,n-1)

a=2, b=n
left>right (2,n) a+=1, b-=1
left>right (3,n-1) 
*/

int main(void){
    char word[] = "settled"; //"adjourned"
    int check = is_cyclone_word(word);
    if (check == 1){
        printf("True\n");
    }
    else 
        printf("False\n");

    // printf("%d\n", word[left]);    
    return 0;
}
